spec/controllers/profile_controller_spec.rb

```diff
@@ -0,0 +1,25 @@
+require "rails_helper"
+
+RSpec.describe ProfileController, :type => :controller do
+
+  describe "GET #index" do
+    context "when logged" do
+      before do
+        @request.env["devise.mapping"] = Devise.mappings[:user]
```

Comment:
Dá uma lida aqui pra tornar esse código reutilizável: https://github.com/plataformatec/devise/wiki/How-To:-Test-controllers-with-Rails-(and-RSpec)